scripts/Diffraction/isis_powder/osiris_routines/osiris_algs.py

```diff
@@ -0,0 +1,300 @@
+    _group_workspaces(focussed, 'OSIRIS' + run_number + WORKSPACE_SUFFIX.focussed)
+    return focussed
+
+
+def get_run_details(run_number_string, inst_settings, is_vanadium_run):
+    all_run_numbers = get_cal_mapping_dict(run_number_string, inst_settings.cal_mapping_path)
+    empty_runs = _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers, key="empty_run_numbers")
+    vanadium_runs = _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers, key="vanadium_run_numbers")
+
+    grouping_file_name = inst_settings.grouping
+
+    spline_identifier = []
+
+    return create_run_details_object(run_number_string=run_number_string, inst_settings=inst_settings,
+                                     is_vanadium_run=is_vanadium_run, splined_name_list=spline_identifier,
+                                     grouping_file_name=grouping_file_name, empty_inst_run_number=empty_runs,
+                                     vanadium_string=vanadium_runs, van_abs_file_name=inst_settings.van_absorb_file)
+
+
+def get_van_runs_for_samples(run_number_string, inst_settings, drange_sets):
+    all_run_numbers = get_cal_mapping_dict(run_number_string, inst_settings.cal_mapping_path)
+    van_numbers = ""
+    for drange_set in drange_sets.values():
+        if drange_set.has_sample():
+            van_numbers += _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers,
+                                                    key="vanadium_" + drange_set.get_drange()) + ","
+    return van_numbers[:-1]
+
+
+def get_empty_runs_for_samples(run_number_string, inst_settings, drange_sets):
+    all_run_numbers = get_cal_mapping_dict(run_number_string, inst_settings.cal_mapping_path)
+    empty_numbers = ""
+    for drange_set in drange_sets.values():
+        if drange_set.has_sample():
+            empty_numbers += _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers,
+                                                      key="empty_" + drange_set.get_drange()) + ","
+    return empty_numbers[:-1]
+
+
+def get_vanadium_runs(inst_settings):
+    all_run_numbers = get_cal_mapping_dict(inst_settings.run_number, inst_settings.cal_mapping_path)
+    return [_assign_drange_vanadium('drange'+str(i+1), all_run_numbers) for i in range(12)]
+
+
+def get_empty_runs(inst_settings):
+    all_run_numbers = get_cal_mapping_dict(inst_settings.run_number, inst_settings.cal_mapping_path)
+    return [_assign_drange_empty('drange'+str(i+1), all_run_numbers) for i in range(12)]
+
+
+def _correct_drange_overlap(merged_ws, drange_sets):
+    # Create scalar data to cope with where merge has combined overlapping data.
+    data_x = (merged_ws.dataX(0)[1:] + merged_ws.dataX(0)[:-1]) / 2.0
+    data_y = np.zeros(data_x.size)
+    for drange in drange_sets:
+        if drange_sets[drange].has_sample():
+            for i in range(data_x.size):
+                if d_range_alice[drange][0] <= data_x[i] <= d_range_alice[drange][1]:
```

Comment:
is alice a slice?

Reply:
Alice is another data reduction process. these values are also from the OSIRIS user manual for cropping the dranges together for a smoother merged output.